plugins/destination/mysql/client/read.go

```diff
@@ -74,7 +74,6 @@ func (*Client) createResultsArray(table *arrow.Schema) []any {
 
 func reverseTransform(table *arrow.Schema, values []any) (arrow.Record, error) {
 	recordBuilder := array.NewRecordBuilder(memory.DefaultAllocator, table)
-	defer recordBuilder.Release()
```

Comment:
I think here it's fine just because the builder is only used in this function scope. we can later try and test if it has any effect on memory vs not using it.

Reply:
The default allocator is the Go one so I thought this is a no op. I can keep it if needed but removed to avoid confusion for someone who's later reading the code

Reply:
Unfortunately `Release` is not a no-op in the Go allocator. But I think it's fine to remove it here and let the garbage collector clean it up for the sake of consistency.

Reply:
Reverted the commit until we reach a conclusion